test/extended/node/oom_detection.go

```diff
@@ -0,0 +1,74 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	o "github.com/onsi/gomega"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: expand this test to SSH into every node in the cluster and gather kubelet logs in case kubelet doesn't send
+// 		 events to the api-server or `oc adm node-logs`
+var _ = g.Describe("[sig-node][Late]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("no-oom-kills")
+
+	g.It("should not have pods that are OOM Killed", func() {
+		kubeClient := oc.AdminKubeClient()
+
+		events, err := kubeClient.CoreV1().Events("").List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			g.Fail(fmt.Sprintf("Unexpected error: %v", err))
+		}
+		oomEventNodeList := []string{}
+		errorMsg := "System OOM encountered"
+		for _, event := range events.Items {
+			if !strings.Contains(event.Message, errorMsg) {
+				continue
+			}
+			oomEventNodeList = append(oomEventNodeList, event.InvolvedObject.Name)
+		}
+		failMessage := fmt.Sprintf("%v OOM events occured in the following nodes\n", oomEventNodeList)
+		if len(oomEventNodeList) > 0 {
+			g.Fail(failMessage)
+			return
+		}
+		// let us see if need to mark this as a flake
+		// result.Flakef(strings.Join(oomEncounteredEvents, "\n"))
+	})
+})
+
+var _ = g.Describe("[sig-node][Late]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("no-oom-kills within worker node journal logs")
+
+	g.It("OOM Killer shouldn't have been invoked in kubelet logs", func() {
+
+		nodeList, err := oc.AdminKubeClient().CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		oomJournalNodeList := []string{}
+		for _, node := range nodeList.Items {
+			// Choosing 150 min as the maximum upgrade duration has been set to 150m in upgrade.go
+			out, err := oc.Run("adm", "node-logs").Args(node.Name, "--path=journal",
```

Comment:
You can get the exact test duration from test util - `DurationSinceStartInSeconds()`